### PR TITLE
[Code health] Inline creation of observables and annotate

### PR DESCRIPTION
**Pull request by gino-m**
Did this for:
- [x] `Subjects`
- [x] `Processors`
- [x] `MutableLiveData`

Replaced `ObservableField` and `ObservableBoolean` with `MutableLiveData` in the process.

Work towards #699.

### CODE REVIEW

**Comment by gino-m** [comment]
@scolsen PTAL?

**Review by scolsen — APPROVED** [review]
lgtm, any idea why the build is failing?

File: gnd/src/androidTest/java/com/google/android/gnd/system/auth/FakeAuthenticationManager.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
   private static final String TAG = FakeAuthenticationManager.class.toString();
 
+  @Hot(replays = true)
```
Comment (scolsen) [inline gnd/src/androidTest/java/com/google/android/gnd/system/auth/FakeAuthenticationManager.java:30]: Missing import for this -- that's why the build failed.

Comment (scolsen) [inline gnd/src/androidTest/java/com/google/android/gnd/system/auth/FakeAuthenticationManager.java:30]: Missing import for this -- that's why the build failed.

**Review by scolsen — CHANGES_REQUESTED** [review]

File: gnd/src/androidTest/java/com/google/android/gnd/system/auth/FakeAuthenticationManager.java (hunk 1 above)
Comment (gino-m) [reply gnd/src/androidTest/java/com/google/android/gnd/system/auth/FakeAuthenticationManager.java:31]: Good catch!!! We were wondering.. I think AS got out of sync somehow, and I didn't realize auto-import didn't work. Thanks for finding this!

**Review by scolsen — DISMISSED** [review]

**Review by scolsen — DISMISSED** [review]

**Comment by scolsen** [comment]
Just needs a merge conflict resolution.

**Review by scolsen — APPROVED** [review]

**Pull request merged**